src/librustc/ty/mod.rs

```diff
@@ -287,6 +287,13 @@ impl Visibility {
     pub fn from_hir(visibility: &hir::Visibility, id: NodeId, tcx: &TyCtxt) -> Self {
         match *visibility {
             hir::Public => Visibility::Public,
+            hir::Visibility::Crate => Visibility::Restricted(ast::CRATE_NODE_ID),
+            hir::Visibility::Restricted { id, .. } => match tcx.def_map.borrow().get(&id) {
+                Some(resolution) => Visibility::Restricted({
+                    tcx.map.as_local_node_id(resolution.base_def.def_id()).unwrap()
+                }),
+                None => Visibility::Public,
```

Comment:
Does compilation continue when a path in `pub(path)` is unresolved? Is `Visibility::Public` used here to avoid spurious privacy errors?